packages/js/src/api/inbox-service.ts

```diff
@@ -27,16 +27,25 @@ export class InboxService {
     subscriberHash,
     subscriber,
   }: {
-    applicationIdentifier: string;
+    applicationIdentifier?: string;
     subscriberHash?: string;
-    subscriber: Subscriber;
+    subscriber?: Subscriber;
   }): Promise<Session> {
-    const response = (await this.#httpClient.post(`${INBOX_ROUTE}/session`, {
-      applicationIdentifier,
-      subscriberHash,
-      subscriber,
-    })) as Session;
+    const response = (await this.#httpClient.post(
+      `${INBOX_ROUTE}/session`,
+      {
+        applicationIdentifier,
+        subscriberHash,
+        subscriber,
+      },
+      {
+        headers: {
+          'Novu-Application-Identifier': applicationIdentifier || 'pk_keyless_',
+        },
```

Comment:
Why is this needed if we dod the setKeylessHeader later?

Reply:
before it was needed for validation of the "keyless environment" , in the new flow it is not needed, so i will remove it.